v2/services/authorize/functions.go

```diff
@@ -32,12 +32,23 @@ func (c *client) SetRequestTimeout(d time.Duration) {
 	c.requestTimeout = d
 }
 
+func (c *client) withRequestTimeout(ctx context.Context) (context.Context, context.CancelFunc) {
+	if _, ok := ctx.Deadline(); !ok {
+		return context.WithTimeout(ctx, c.requestTimeout)
+	}
+
+	return ctx, func() {}
+}
```

Comment:
Does this really work as intended in the case where the deadline has passed?

Reply:
Not sure I understand the concern. We add a default deadline in case no deadline has been set by the caller.